aten/src/ATen/native/TensorShape.cpp

```diff
@@ -2171,6 +2171,13 @@ std::vector<Tensor> meshgrid(TensorList tensors) {
 // with dims reversed
 Tensor numpy_T(const Tensor &self) {
   int64_t n = self.dim();
+  if (n != 2) {
+    TORCH_WARN_ONCE(
+        "The use of `x.T` on tensors of dimension other than 2 is deprecated and will throw ",
+        "an error in a future release. Consider `x.mT` to transpose batches of matrices or ",
```

Comment:
It's probably worth saying why it's deprecated, aka because it probably doesn't do what you want when things are batched